fluent/syntax/parser.py

```diff
@@ -270,10 +277,11 @@ def get_variant_key(self, ps):
         if ch is None:
             raise ParseError('E0013')
 
-        if ps.is_number_start():
+        cc = ord(ch)
+        if ((cc >= 48 and cc <= 57) or cc == 45):  # 0-9, -
```

Comment:
`ps.is_number_start` sounds like the right choice here. Maybe we should fix the JS impl instead of removing it here?

Reply:
(I can do it in `fluent.js` if you decide to change it here.)

Reply:
I don't know why but without this change it was breaking some tests. I didn't have time to investigate. Happy to port a patch if you write one for .js

Reply:
OK, let's leave it as-is for now, thanks!